src/patternfly/components/Drawer/drawer.scss

```diff
@@ -21,55 +33,195 @@
+  order: 0;
+}
 
-    @media screen and (min-width: $pf-global--breakpoint--md) {
-      &.pf-m-inline {
-        --pf-c-drawer__content--MarginRight: var(--pf-c-drawer--m-expanded--m-inline__content--MarginRight);
+// Panel
+.pf-c-drawer__panel {
+  position: relative;
+  flex-basis: var(--pf-c-drawer__panel--FlexBasis);
+  order: 1;
+  background-color: var(--pf-c-drawer__panel--BackgroundColor);
+  transition-duration: var(--pf-c-drawer__panel--TransitionDuration);
+  transition-property: var(--pf-c-drawer__panel--TransitionProperty);
+  -webkit-overflow-scrolling: touch;
+}
+
+// Content body
+.pf-c-drawer__content-body.pf-m-padding {
+  padding: var(--pf-c-drawer__content-body--m-padding--PaddingTop) var(--pf-c-drawer__content-body--m-padding--PaddingRight) var(--pf-c-drawer__content-body--m-padding--PaddingBottom) var(--pf-c-drawer__content-body--m-padding--PaddingLeft);
+}
+
+// Panel body
+.pf-c-drawer__panel-body {
+  padding: var(--pf-c-drawer__panel-body--PaddingTop) var(--pf-c-drawer__panel-body--PaddingRight) var(--pf-c-drawer__panel-body--PaddingBottom) var(--pf-c-drawer__panel-body--PaddingLeft);
+}
+
+// Panel body
+.pf-c-drawer__panel-body.pf-m-no-padding {
+  --pf-c-drawer__panel-body--PaddingTop: 0;
+  --pf-c-drawer__panel-body--PaddingRight: 0;
+  --pf-c-drawer__panel-body--PaddingBottom: 0;
+  --pf-c-drawer__panel-body--PaddingLeft: 0;
+}
+
+// Expanded
+.pf-c-drawer.pf-m-expanded .pf-c-drawer__panel {
+  transform: translateX(-100%);
+}
+
+// Layout modifications happen after md breakpoint
+@media screen and (min-width: $pf-global--breakpoint--md) {
+  // Default
+  .pf-c-drawer:not(.pf-m-divider) .pf-c-drawer__panel {
+    box-shadow: var(--pf-c-drawer__panel--BoxShadow);
+  }
+
+  .pf-c-drawer {
+    .pf-c-drawer__content {
+      order: 0;
+    }
+
+    .pf-c-drawer__panel {
+      order: 1;
+      transform: translateX(0);
+    }
+
+    // Expanded
+    &.pf-m-expanded {
+      .pf-c-drawer__panel {
+        --pf-c-drawer__panel--BoxShadow: var(--pf-c-drawer--m-expanded__panel--BoxShadow);
+
+        transform: translateX(-100%);
       }
     }
+  }
+
+  // Panel left
+  .pf-c-drawer.pf-m-panel-left {
+    .pf-c-drawer__content {
+      order: 1;
+    }
 
     .pf-c-drawer__panel {
-      box-shadow: var(--pf-c-drawer--m-expanded__panel--BoxShadow);
+      order: 0;
+      margin-right: calc(var(--pf-c-drawer__panel--FlexBasis) * -1);
+      transform: translateX(-100%);
+    }
+
+    // Expanded, panel left
+    &.pf-m-expanded .pf-c-drawer__panel {
+      --pf-c-drawer__panel--BoxShadow: var(--pf-c-drawer--m-expanded--m-panel-left__panel--BoxShadow);
+
+      transform: translateX(0);
     }
   }
 }
 
-.pf-c-drawer__content {
-  z-index: var(--pf-c-drawer__content--ZIndex);
-  flex-grow: 1;
-  min-width: 0;
-  margin-right: var(--pf-c-drawer__content--MarginRight);
-  transition: var(--pf-c-drawer__content--Transition);
+// Drawer inline
+@media screen and (min-width: $pf-global--breakpoint--md) {
```

Comment:
could you remove this media query and just move the contents into the previous one?